core/exec.c

```diff
@@ -963,7 +963,59 @@ static lai_api_error_t lai_exec_reduce_op(int opcode, lai_state_t *state, struct
         lai_operand_emplace(state, &operands[2], &result);
         break;
     }
+    case MID_OP: {
+        lai_variable_t object = {0};
+        lai_variable_t index = {0};
+        lai_variable_t length = {0};
+        lai_exec_get_objectref(state, &operands[0], &object);
+        lai_exec_get_integer(state, &operands[1], &index);
+        lai_exec_get_integer(state, &operands[2], &length);
+
+        size_t strl = 0;
+        size_t n = index.integer;
+        size_t sz = length.integer;
+        if (object.type == LAI_STRING) {
+            strl = lai_exec_string_length(&object);
+        } else if (object.type == LAI_BUFFER) {
+            strl = lai_exec_buffer_size(&object);
+        }
 
+        if (n >= strl) {
+            sz = 0;
+        } else if ((n + sz) > strl) {
+            sz = strl - n;
+        }
+
+        switch (object.type) {
+            case LAI_STRING: {
```

Comment:
Does MidOp return a buffer for a string?

Reply:
It does return a string for a string, fixed it.

Reply:
It does return a string for a string, fixed it.